synfig-core/src/synfig/threadpool.cpp

```diff
@@ -275,6 +275,21 @@ ThreadPool::enqueue(const Slot &slot) {
 	wakeup();
 }
 
+
+void 
+ThreadPool::set_threads(int no_core){
+	max_running_threads = g_get_num_processors();
+	if(no_core!=0){
+		max_running_threads = no_core;
+	}
+	if (const char *s = getenv("SYNFIG_GENERIC_THREADS"))
+		max_running_threads = atoi(s) + 1;
+
+	if (max_running_threads < 2) max_running_threads = 2;
+	if (max_running_threads > 2) --max_running_threads;
+	++running_threads;
+}
+
```

Comment:
Maybe the `max_running_threads`should be bounded to num_processors, even if 'no_core' is greater than it.